Sources/SourceKitLSP/Swift/DoccDocumentation.swift

```diff
@@ -0,0 +1,156 @@
+    guard
+      let nearestDocumentableSymbol = DocumentableSymbol.findNearestSymbol(
+        syntaxTree: syntaxTree,
+        position: snapshot.absolutePosition(of: position)
+      )
+    else {
+      throw ResponseError.requestFailed(doccDocumentationError: .noDocumentation)
+    }
+    // Retrieve the symbol graph as well as information about the symbol
+    let symbolPosition = await adjustPositionToStartOfIdentifier(
+      snapshot.position(of: nearestDocumentableSymbol.position),
+      in: snapshot
+    )
+    let (cursorInfo, _, symbolGraph) = try await cursorInfo(
+      req.textDocument.uri,
+      Range(symbolPosition),
+      includeSymbolGraph: true,
+      fallbackSettingsAfterTimeout: false
+    )
+    guard let symbolGraph,
+      let cursorInfo = cursorInfo.first,
+      let symbolUSR = cursorInfo.symbolInfo.usr
+    else {
+      throw ResponseError.internalError("Unable to retrieve symbol graph for the document")
+    }
+    // Locate the documentation extension and include it in the request if one exists
+    let markupExtensionFile = try? await findMarkupExtensionFile(
+      workspace: workspace,
+      documentationManager: documentationManager,
+      catalogURL: catalogURL,
+      for: symbolUSR
+    )
+    return try await documentationManager.renderDocCDocumentation(
+      symbolUSR: symbolUSR,
+      symbolGraph: symbolGraph,
+      overrideDocComments: nearestDocumentableSymbol.documentationComments,
+      markupFile: markupExtensionFile,
+      moduleName: moduleName,
+      catalogURL: catalogURL
+    )
+  }
+
+  private func findMarkupExtensionFile(
+    workspace: Workspace,
+    documentationManager: DocCDocumentationManager,
+    catalogURL: URL?,
+    for symbolUSR: String
+  ) async throws -> String? {
+    guard let catalogURL else {
+      return nil
+    }
+    let catalogIndex = try await documentationManager.catalogIndex(for: catalogURL)
+    guard let index = workspace.index(checkedFor: .deletedFiles),
+      let symbolInformation = DocCSymbolInformation(fromUSR: symbolUSR, in: index),
+      let markupExtensionFileURL = catalogIndex.documentationExtension(for: symbolInformation)
+    else {
+      return nil
+    }
+    return try? documentManager.latestSnapshotOrDisk(
+      DocumentURI(markupExtensionFileURL),
+      language: .markdown
+    )?.text
+  }
+}
+
+fileprivate struct DocumentableSymbol {
+  let position: AbsolutePosition
+  let documentationComments: [String]
+
+  init(node: any SyntaxProtocol, position: AbsolutePosition) {
+    self.position = position
+    self.documentationComments = node.leadingTrivia.flatMap { trivia -> [String] in
+      switch trivia {
+      case .docLineComment(let comment):
+        return [String(comment.dropFirst(3).trimmingCharacters(in: .whitespaces))]
+      case .docBlockComment(let comment):
+        return comment.dropFirst(3)
+          .dropLast(2)
+          .split(separator: "\n")
```

Comment:
We should also split on `\r\n`, probably using `.split(whereSeparator: \.isNewline)`

Also, how does this handle doc comments like

```swift
/* 
 * abc
 * def
 */
func test() {}
```

Or does docc not handle this style where the doc lines are prepended by `*` at all (in which case it’s fine)?